lib/connection/connection_config.js

```diff
@@ -481,6 +481,26 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo) {
     disableConsoleLogin = options.disableConsoleLogin;
   }
 
+  let numofRetries = 1;
+
+  this.getNumofRetries = function () {
+    return numofRetries;
+  };
+
+  this.setNumofRetries = function (retry) {
+    numofRetries = retry;
+  };
+
+  let elapsedTime = 0;
+
+  this.getElapsedTime = function () {
```

Comment:
I don't like this approach since elapsedTime is per request while connection_config is on connection level.
Can we expend reauthenticate() interface to pass elapsed time or maybe even retried attempts there?

Reply:
I fixed it and passed the retryOption as a parameter to the re-authenticate function. Please have a look.